### PR TITLE
Vmem pte api

### PR DESCRIPTION
**Pull request by anton-bondarev**
* Rework vmem_translate()
* Improve some mmu commands
* Add impls for new vmem API for (x86, microblaze, arm, sparc)

### CODE REVIEW

File: src/arch/arm/mmu/mmu_section.c [hunk 1]
```diff
@@ -13,76 +14,104 @@
...
+
+	if (prot & PROT_EXEC) {
+		flags &= ~L1D_XN; /* ARM_MMU_SECTION_XN */
 	} else {
-		*pte |= L1D_C; /* ARM_MMU_SECTION_C */
+		flags |= L1D_XN; /* ARM_MMU_SECTION_XN */
 	}
+	return (addr & ~MMU_PAGE_MASK) | flags | ARM_MMU_TYPE_SECTION;
+#endif
+	return ((addr & ~MMU_PAGE_MASK)
```
Comment (Deryugin) [inline src/arch/arm/mmu/mmu_section.c]: So it always maps memory as uncached?

File: src/arch/microblaze/mmu/mmuinfo.c [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+int arch_mmu_get_regs_table(struct mmuinfo_regs *buf, int buf_size, int offset) {
+	int cnt;
+	int i;
+
+	cnt = ARRAY_SIZE(mmuinfo_regs) - offset;
+	if (cnt <= 0) {
+		return 0;
+	}
+	cnt = min(buf_size, cnt);
+	for (i= 0; i < cnt; i ++) {
```
Comment (Deryugin) [inline src/arch/microblaze/mmu/mmuinfo.c]: `(i= 0` -> `(i = 0`

File: src/arch/sparc/mmu/mmuinfo.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+int arch_mmu_get_regs_table(struct mmuinfo_regs *buf, int buf_size, int offset) {
+	int cnt;
+	int i;
+
+	cnt = ARRAY_SIZE(mmuinfo_regs) - offset;
+	if (cnt <= 0) {
+		return 0;
+	}
+	cnt = min(buf_size, cnt);
+	for (i= 0; i < cnt; i ++) {
```
Comment (Deryugin) [inline src/arch/sparc/mmu/mmuinfo.c]: `for (i= 0`->`for (i = 0`

Comment (Deryugin) [reply src/arch/sparc/mmu/mmuinfo.c]: `i ++`->`i++`

File: src/arch/microblaze/mmu/mmuinfo.c (hunk 2 above)
Comment (Deryugin) [reply src/arch/microblaze/mmu/mmuinfo.c]: `i ++`->`i++`

File: src/arch/x86/mmu/mmuinfo.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+int arch_mmu_get_regs_table(struct mmuinfo_regs *buf, int buf_size, int offset) {
+	int cnt;
+	int i;
+
+	cnt = ARRAY_SIZE(mmuinfo_regs) - offset;
+	if (cnt <= 0) {
+		return 0;
+	}
+	cnt = min(buf_size, cnt);
+	for (i= 0; i < cnt; i ++) {
```
Comment (Deryugin) [inline src/arch/x86/mmu/mmuinfo.c]: `for (i= 0`->`for (i = 0`
`i ++`->`i++`

File: src/mem/vmem/vmem_nommu.h [hunk 5]
```diff
@@ -17,6 +17,6 @@
 
 #define vmem_free_context(ctx)
 
-#define vmem_translate(a,b) b
+#define vmem_translate(a, b, c) b
```
Comment (Deryugin) [inline src/mem/vmem/vmem_nommu.h]: `#define vmem_translate(a, b, c) (b)`

File: src/arch/arm/mmu/mmu_section.c (hunk 1 above)
Comment (anton-bondarev) [reply src/arch/arm/mmu/mmu_section.c]: Yes.I didn't change logic, just API

File: src/arch/microblaze/mmu/mmuinfo.c (hunk 2 above)
Comment (anton-bondarev) [reply src/arch/microblaze/mmu/mmuinfo.c]: done

File: src/arch/sparc/mmu/mmuinfo.c (hunk 3 above)
Comment (anton-bondarev) [reply src/arch/sparc/mmu/mmuinfo.c]: done

File: src/arch/x86/mmu/mmuinfo.c (hunk 4 above)
Comment (anton-bondarev) [reply src/arch/x86/mmu/mmuinfo.c]: done

File: src/mem/vmem/vmem_nommu.h (hunk 5 above)
Comment (anton-bondarev) [reply src/mem/vmem/vmem_nommu.h]: done

**Pull request merged**